signac/core/syncedattrdict.py

```diff
@@ -17,10 +17,48 @@
 from ..errors import KeyTypeError
 
 
+class AttrDictFormatValidator:
```

Comment:
I don't think this class needs to exist. There's no case (at least right now) where the `JSONFormatValidator` wouldn't be what we wanted to use in all cases. Also in practice (although we don't have to impose this restriction anywhere in the API) I expect that validation is something that will be used for compatibility with backend specifications, not for filtering specific data types (although that's certainly possible). Also, as I said in a previous comment the validators and collections should be more independent, so the `SyncedAttrDict` doesn't need any validators hardcoded into its initializer anyway.

Reply:
I will change this to the old implementation (`_valiadate_key`).

Reply:
I'm not suggesting that, I'm suggesting to use the JSONFormatValidator here. I don't see any reason that you can't. The JSONFormatValidator is a superset of this one that also works for non-mapping data types, but is otherwise doing the same thing, right?